openbook_common/helpers.py

```diff
@@ -30,3 +38,92 @@ def get_supported_translation_language(language_code):
     supported_translation_code = translation_strategy.get_supported_translation_language_code(language_code)
 
     return Language.objects.get(code=supported_translation_code)
+
+
+def get_matched_urls_from_text(text):
+    """
+    Returns all the raw extracted urls as a list
+    If a URL has a scheme, it ensures that it is https
+    URLs like www. are sanitised in the get_sanitised_url_for_link
+    """
+    text = text.lower()
+    extractor = URLExtract()
+    results = [url for url in extractor.gen_urls(text)]
+    for url in results:
+        scheme = urlparse(url).scheme
+        if scheme and scheme != 'https':
```

Comment:
So this won't work with things like techcrunch.com ?

Reply:
It will, only if the scheme is present. To avoid weird protocols like `file://` to be used, for sites like techcrunch.com, www.techcrunch.com, we will rely on the fact that we will whitelist only reliable domains.

Reply:
But if there's a http:// scheme, it should be valid...